clients/src/main/java/org/apache/kafka/common/serialization/SerDe.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * The interface for wrapping a serializer and deserializer for the given data type.
+ *
+ * @param <T>
+ */
+public interface SerDe<T> {
```

Comment:
See [my comment below](https://github.com/apache/kafka/pull/1066/files#r56299007) on naming consistency for `SerDe` vs. `Serde`.  I don't have a strong opinion which variant we go with, but I do like us to be consistent.

Personally, I'd slightly favor `Serde` if only because it is easier to type (most people, e.g. in docs or in mailing lists, will write "serde" or "serdes" anyway...). :-)